ir_x86.dasc

```diff
@@ -11541,9 +11549,17 @@ next_block:;
 		return NULL;
 	}
 
+	if (data.dasm_state->status != DASM_S_OK) {
+		IR_ASSERT(data.dasm_state->status == DASM_S_NOMEM);
+		dasm_free(&data.dasm_state);
+		ctx->data = NULL;
+		ctx->status = IR_ERROR_TOO_LARGE;
```

Comment:
For RELEASE build (when IR_ASSERT is removed) ctx->status might be incorrect.
Please check all the similar places.